src/modeler/AutoRest.Swagger/Validation/TrackedResourceListByImmediateParent.cs

```diff
@@ -35,23 +35,83 @@ public class TrackedResourceListByImmediateParent : TypedRule<Dictionary<string,
+         *      Step 2: /subscriptions/{subscriptionId}/resourceGroup/{resourceGroupName}/providers/Microsoft.Sql/servers/{server1}/databases
+         *      Step 3: databases
+         */
+        private static readonly Regex resourcePathPattern = new Regex("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/[^/]+/[^/]+/{[^/]+}.*/(?<childresource>\\w+)/{[^/]+}$", RegexOptions.IgnoreCase);
+
         private KeyValuePair<string, string> GetChildAndImmediateParentResource(string path, Dictionary<string, Dictionary<string, Operation>> paths, Dictionary<string, Schema> definitions)
         {
-            Match match = ValidationUtilities.ResourcePathPattern.Match(path);
+            Match match = resourcePathPattern.Match(path);
             KeyValuePair<string, string> result = new KeyValuePair<string, string>();
-            // Look for unparameterized resource. If it is not available, pick the last resource provided it has a parent resource
-            if (match.Success && (match.Groups["unparameterizedresource"].Success || (match.Groups["resource"].Success && match.Groups["resource"].Captures.Count > 1)))
+            if (match.Success)
             {
-                int previousResourceLocation = match.Groups["resource"].Captures.Count - 1;
-                string childResourceName = match.Groups["unparameterizedresource"].Success? match.Groups["unparameterizedresource"].Value: match.Groups["resource"].Captures[previousResourceLocation].Value;
-                string immediateParentResourceNameInPath = match.Groups["unparameterizedresource"].Success? match.Groups["resource"].Captures[previousResourceLocation].Value : match.Groups["resource"].Captures[previousResourceLocation - 1].Value;
+                string childResourceName = match.Groups["childresource"].Value;
+                string immediateParentResourceNameInPath = GetImmediateParentResourceName(path);
                 string immediateParentResourceNameActual = GetActualParentResourceName(immediateParentResourceNameInPath, paths, definitions);
                 result = new KeyValuePair<string, string>(childResourceName, immediateParentResourceNameActual);
             }
 
             return result;
         }
 
+        /// <summary>
+        /// Gets the immediate parent resource name
+        /// </summary>
+        /// <param name="childResourcePaths">paths that fits the child resource criteria</param>
+        /// <returns>name of the immediate parent resource name</returns>
+        private string GetImmediateParentResourceName(string pathToEvaluate)
+        {
+            pathToEvaluate = pathToEvaluate.Substring(0, pathToEvaluate.LastIndexOf("/{"));
+            pathToEvaluate = pathToEvaluate.Substring(0, pathToEvaluate.LastIndexOf("/{"));
+            return pathToEvaluate.Substring(pathToEvaluate.LastIndexOf("/") + 1);
```

Comment:
@dsgouda  What do you mean by "if there is only child resource in the path"? Can you give an example?